sparse/impl/KokkosSparse_sptrsv_cuSPARSE_impl.hpp

```diff
@@ -60,6 +60,142 @@ void sptrsvcuSPARSE_symbolic(KernelHandle* sptrsv_handle,
+  typedef typename KernelHandle::memory_space memory_space;
+  typedef typename KernelHandle::nnz_scalar_view_t nnz_scalar_view_t;
+
+  const bool is_cuda_space =
+      std::is_same<memory_space, Kokkos::CudaSpace>::value ||
+      std::is_same<memory_space, Kokkos::CudaUVMSpace>::value ||
+      std::is_same<memory_space, Kokkos::CudaHostPinnedSpace>::value;
+
+  const bool is_idx_type_supported = std::is_same<idx_type, int>::value ||
+                                     std::is_same<idx_type, int64_t>::value;
+
+  if (!is_cuda_space) {
+    throw std::runtime_error(
+        "KokkosKernels sptrsvcuSPARSE_symbolic: MEMORY IS NOT ALLOCATED IN GPU "
+        "DEVICE for CUSPARSE\n");
+  } else if (is_idx_type_supported) {
+    bool is_lower = sptrsv_handle->is_lower_tri();
+    sptrsv_handle->create_cuSPARSE_Handle(trans, is_lower);
+
+    typename KernelHandle::SPTRSVcuSparseHandleType* h =
+        sptrsv_handle->get_cuSparseHandle();
+
+    cusparseStatus_t status;
+
+    int64_t nnz = static_cast<int64_t>(entries.extent(0));
+    size_t pBufferSize;
+    void* rm;
+    // NOTE (Oct-29-2022):
+    // cusparseCreateCsr only supports the same sizes (either 32 bits or 64
+    // bits) for row_map_type and entries_type
+    if (std::is_same<idx_type, int>::value) {
+      if (!std::is_same<size_type, int>::value)
+        sptrsv_handle->allocate_tmp_int_rowmap(row_map.extent(0));
+      rm = !std::is_same<size_type, int>::value
+               ? (void*)sptrsv_handle->get_int_rowmap_ptr_copy(row_map)
+               : (void*)row_map.data();
+    } else {  // idx_type has 64 bits
+      if (!std::is_same<size_type, int64_t>::value)
+        sptrsv_handle->allocate_tmp_int64_rowmap(row_map.extent(0));
+      rm = !std::is_same<size_type, int64_t>::value
+               ? (void*)sptrsv_handle->get_int64_rowmap_ptr_copy(row_map)
+               : (void*)row_map.data();
+    }
+    const scalar_type alpha = scalar_type(1.0);
+
+    cusparseIndexType_t cudaCsrColIndType, cudaCsrRowMapType;
+    cudaDataType cudaValueType;
+
+    if (std::is_same<idx_type, int>::value) {
+      cudaCsrColIndType = CUSPARSE_INDEX_32I;
+      cudaCsrRowMapType = CUSPARSE_INDEX_32I;
+    } else {  // idx_type has 64 bits
+      cudaCsrColIndType = CUSPARSE_INDEX_64I;
+      cudaCsrRowMapType = CUSPARSE_INDEX_64I;
+    }
+
+    if (std::is_same<scalar_type, double>::value) {
+      cudaValueType = CUDA_R_64F;
+    } else if (std::is_same<scalar_type, float>::value) {
+      cudaValueType = CUDA_R_32F;
+    } else if (std::is_same<scalar_type, Kokkos::complex<double>>::value) {
+      cudaValueType = CUDA_C_64F;
+    } else if (std::is_same<scalar_type, Kokkos::complex<float>>::value) {
+      cudaValueType = CUDA_C_32F;
+    } else {
+      throw std::runtime_error("CUSPARSE wrapper error: unsupported type.\n");
+    }
+
+    // Create sparse matrix in CSR format
+    status = cusparseCreateCsr(&(h->matDescr), static_cast<int64_t>(nrows),
```

Comment:
It would be nice to wrap this with [KOKKOS_CUSPARSE_SAFE_CALL](https://github.com/kokkos/kokkos-kernels/blob/491fbfba3c183c5d7ba2b2eb5fdd1e3a9e2a8bac/sparse/src/KokkosSparse_Utils_cusparse.hpp#L115)

Reply:
Done